app/src/androidTest/java/ch/epfl/sdp/blindly/match/ProfileTest.kt

```diff
@@ -0,0 +1,69 @@
+package ch.epfl.sdp.blindly.match
+
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.hamcrest.Matchers.contains
+import org.hamcrest.Matchers.equalTo
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@HiltAndroidTest
+class ProfileTest {
+
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+    }
+
+    val andre1 = Profile(name = "André", age =  25)
+    val andre2 = Profile(name = "André", age = 24)
+    val pierre1 = Profile(name = "Pierre", age = 24)
+    val pierre2 = Profile(name = "Pierre", age = 25)
+
+    @Test
+    fun equalsIsTrueWithSameProfile() {
+        assertThat(andre1 == andre1, equalTo(true))
+    }
+
+    @Test
+    fun equalsIsFalseWithDifferentProfiles() {
+        assertThat(andre1 == pierre1, equalTo(false))
+    }
+
+    @Test
+    fun equalsIsFalseForDifferentNames() {
+        assertThat(andre1 == pierre2, equalTo(false))
+    }
+
+    @Test
+    fun equalsIsFalseForDifferentAges() {
+        assertThat(andre2 == pierre1, equalTo(false))
```

Comment:
Wouldn't better to test andre1 and andre2 here ? So that the name is the same but not the equality ?

Reply:
That's true, I agree ! Otherwise the name of the test doesn't reflect what it does.

Reply:
fixed it